src/cmd/src/cli/export.rs

```diff
@@ -328,23 +348,26 @@ impl Export {
                     Path::new(&self.output_dir).join(format!("{catalog}-{schema}_copy_from.sql"));
                 let mut file = File::create(copy_from_file).await.context(FileIoSnafu)?;
 
-                let copy_from_sql = dir_filenames
-                    .into_iter()
-                    .map(|file| {
-                        let file = file.unwrap();
-                        let filename = file.file_name().into_string().unwrap();
+                for table_file in dir_filenames {
+                    let table_file = table_file.unwrap();
+                    let table_name = table_file
+                        .file_name()
+                        .into_string()
+                        .unwrap()
+                        .replace(".parquet", "");
 
+                    file.write(
                         format!(
                             "copy {} from '{}' with (format='parquet');\n",
-                            filename.replace(".parquet", ""),
-                            file.path().to_str().unwrap()
+                            table_name,
+                            table_file.path().to_str().unwrap()
                         )
-                    })
-                    .collect::<Vec<_>>()
-                    .join("");
-                file.write_all(copy_from_sql.as_bytes())
+                        .as_bytes(),
+                    )
                     .await
                     .context(FileIoSnafu)?;
+                    file.flush().await.context(FileIoSnafu)?;
+                }
```

Comment:
We can move flush outside of the for loop.
```suggestion
                }
                file.flush().await.context(FileIoSnafu)?;
```